Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -2315,6 +2386,25 @@ void CGContextClearRect(CGContextRef context, CGRect rect) {
         this->PopEndDraw();
     });
 
+    // If the context has requested antialiasing other than the defaults we now need to update the device context.
+    if (allowsAntialiasing && CurrentGState().shouldAntialias != _kCGTrinaryDefault) {
+        deviceContext->SetAntialiasMode(GetAntialiasMode());
+
+        // If any text rendering parameters have been updated, we need to update the device context.
+        if ((CurrentGState().shouldSmoothFonts != _kCGTrinaryDefault && allowsFontSmoothing) ||
+            (CurrentGState().shouldSubpixelPosition != _kCGTrinaryDefault && allowsFontSubpixelPositioning) ||
+            (CurrentGState().shouldSubpixelQuantizeFonts != _kCGTrinaryDefault && allowsFontSubpixelQuantization)) {
+            ComPtr<IDWriteRenderingParams> originalTextRenderingParams;
+            deviceContext->GetTextRenderingParams(&originalTextRenderingParams);
+
+            ComPtr<IDWriteRenderingParams> customParams;
+            GetTextRenderingParams(&originalTextRenderingParams, &customParams);
```

Comment:
Actually, this is bad. & calls ReleaseAndGetAddressOf.
We should make it not take a pointer to the original params; it can simply create new ones on the fly for the custom params.